docs/features/multidomain.rst

```diff
@@ -0,0 +1,333 @@
+dedicated layer 2 network each, which are interconnected via their
+gateways by layer 3 routing. Gluon is already field-tested handling a
+single domain and the multidomain feature allows for the reconfiguration
+of key parameters that decide which domain a node participates in.
+
+Overview
+--------
+
+Mutlidomain support allows to build a single firmware with multiple,
+switchable domain configurations. The nomenclature is as follows:
+
+-  ``site``: an aggregate over multiple domains
+-  ``domain``: mesh network with connectivity parameters that preventing
+   accidental bridging with other domains
+-  ``domain code``: unique domain identifier
+-  ``domain name``: pretty name for a domain code
+
+By default Gluon builds with a single domain embedded into
+``site.conf``. To use multiple domains, enable it in ``site.mk``:
+
+::
+
+    GLUON_MULTIDOMAIN=1
+
+In the site repository, create the ``domains/`` directory, which will
+hold your domain configurations. Each domain configuration file is named
+after its primary ``domain_code``, additional domain codes and names are
+supported.
+
+::
+
+    site/
+    |-- site.conf
+    |-- site.mk
+    |-- i18n/
+    |-- domains/
+      |-- alpha_centauri.conf
+      |-- beta_centauri.conf
+      |-- gamma_centauri.conf
+
+The domain configuration ``alpha_centauri.conf`` could look like this.
+
+::
+
+    {
+      domain_names = {
+        alpha_centauri = 'Alpha Centauri'
+      },
+      
+      -- more domain specific config follows below
+    }
+
+In this example “Alpha Centauri” is the pretty ``domain_name`` for the
+domain_code ``alpha_centauri``. Also note that the domain code
+``alpha_centauri`` matches the filename ``alpha_centauri.conf``.
+
+Additional domain codes/names can be added to ``domain_names``, which
+are treated as aliases for the their domain configuration. Aliases can
+be used to offer more fine-grained and better recognizable domain
+choices to users. Having multiple aliases on a single domain are helpful
+precursors to splitting the domain into smaller blocks. Aliases also
+prevent the need to duplicate the domain configuration, instead they get
+symbollicaly linked next to existing domain configurations on nodes.
+
+Switching the domain
+--------------------
+
+**via commandline**:
+
+::
+
+    uci set gluon.core.domain="newdomaincode"
+    gluon-reconfigure
+    reboot
+
+**via config mode:**
+
+To allow switching the domain via config mode, the package
+``gluon-config-mode-domain-select`` has to be added into the site.mk.
+
+|image0|
+
+Allowed site variables
+----------------------
+
+Internally the site variables are merged from the ``site.conf`` and the
+selected ``domain.conf``, so the most variables are also allowed in
+``site.conf`` and in ``domain.conf``. But there are some exceptions,
+which do not make sense in a domain or site specific way. In the
+following sections the restricted site variables are listed and sorted
+by the reason why they are restricted.
```

Comment:
The following sections gives an overview over variables that are only usable in either the site or domain context.

Reply:
correction of the correction: "sections give an"